### DISCUSSION TITLE
TCP Client ESP32-EVB PlatformIO

### DISCUSSION THREAD

**Discussion by chrbru** [eModbus/eModbus#266] [General]
Hi, everyone!
I have a Olimex ESP32-EVB revision I board, means with wifi via the espressif chip and an external LAN8710.
I installed the latest version of eModbus Library 1.6.0. This automatically install also the customized Ethernet library that is mentioned in the eModbus documention. But this customized Ethernet library use "only" wifi chips W5xxx (I guess this is an external chip, but not sure).
I use the TCP03example2 as to setup a TCP client.
I can compile the project, but  the LAN and Wifi chips are not detected by the software. Result: An IP connection never get active, because the Wifi/LAN Chip is not detected and then the rest (max amount of sockets, etc.) will never get valid.
Can someone give me a hint what I need to change to make it run, please.

**Comment by Miq1** [comment]
You will need to find a library matching your Ethernet board that provides the standard API calls like connect, open etc.
